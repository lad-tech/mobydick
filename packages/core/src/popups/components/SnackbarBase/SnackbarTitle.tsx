import {FC} from 'react';

import useStyles from '../../../styles/hooks/useStyles';
import {TypographyLegacy} from '../../../typography/components/TypographyLegacy/TypographyLegacy';
import {ITitlePopup} from '../PopupBase';

import stylesCreate from './stylesCreate';

const SnackbarTitle: FC<ITitlePopup> = props => {
  const [styles] = useStyles(stylesCreate);
  const {title, titleStyles, titleFont} = props;

  return (
    <TypographyLegacy
      style={[styles.title, titleStyles]}
      font={titleFont ? titleFont : 'Medium-Contrast-XS'}>
      {title}
    </TypographyLegacy>
  );
};

export default SnackbarTitle;
