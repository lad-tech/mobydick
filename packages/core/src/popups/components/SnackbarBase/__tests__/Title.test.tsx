import {render} from '@testing-library/react-native';

import SnackbarTitle from '../SnackbarTitle';

describe('@lad-tech/mobydick-core/SnackbarBase/Title', () => {
  it('should renders correctly without titleFont', () => {
    const {toJSON} = render(
      <SnackbarTitle title={'title'} titleStyles={{flex: 1}} />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
  it('should renders correctly with titleFont', () => {
    const {toJSON} = render(
      <SnackbarTitle
        title={'title two'}
        titleStyles={{backgroundColor: '#000'}}
        titleFont={'Medium-Secondary-M'}
      />,
    );

    expect(toJSON()).toMatchSnapshot();
  });
});
