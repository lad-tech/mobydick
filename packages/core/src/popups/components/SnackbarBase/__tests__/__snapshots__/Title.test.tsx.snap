// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@lad-tech/mobydick-core/SnackbarBase/Title should renders correctly with titleFont 1`] = `
<Text
  allowFontScaling={true}
  style={
    [
      {
        "color": "#353A45",
        "fontFamily": "Inter-Medium",
        "fontSize": 16,
        "fontWeight": "500",
        "lineHeight": 20,
        "minHeight": 20,
      },
      [
        {
          "flex": 1,
          "paddingRight": 8,
        },
        {
          "backgroundColor": "#000",
        },
      ],
    ]
  }
>
  title two
</Text>
`;

exports[`@lad-tech/mobydick-core/SnackbarBase/Title should renders correctly without titleFont 1`] = `
<Text
  allowFontScaling={true}
  style={
    [
      {
        "color": "#FFFFFF",
        "fontFamily": "Inter-Medium",
        "fontSize": 14,
        "fontWeight": "500",
        "lineHeight": 18,
        "minHeight": 18,
      },
      [
        {
          "flex": 1,
          "paddingRight": 8,
        },
        {
          "flex": 1,
        },
      ],
    ]
  }
>
  title
</Text>
`;
